lib/inherited_proofing/mhv/service.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'inherited_proofing/mhv/configuration'
+
+module InheritedProofing
+  module MHV
+    # This class will interact with MHV and return a hash of user data.
+    # If the user is eligible for inherited proofing the result will be a
+    # hash of identity data with an included code we will cache and send
+    # to login.gov. If the user is not eligible the result will be a hash
+    # of error information. If the entire service fails for whatever
+    # reason the result will be an empty hash.
+    class Service < Common::Client::Base
+      configuration InheritedProofing::MHV::Configuration
+
+      attr_reader :user
+
+      def initialize(user)
+        @user = user
+      end
+
+      def identity_proof_data
+        return {} if correlation_id.blank?
+
+        data_hsh = mhv_api_request(verification_info_url)
+        data_hsh['identityDocumentExist'].present? ? data_hsh.merge(success_attributes) : data_hsh
```

Comment:
I think we want to either shove all relevant logic into this service (which means this service would deal with creating the redis object and would return only the code, or nil if identity document does not exist), or we want to remove any logic beyond just calling the service into a service object pattern.

My personal preference would be a service object pattern, as then we'd have a reusable service to mhv with clear inputs and outputs, and a simple service object that controlled the service calls and saving to the redis. It would look like this:

`MHV::Service` (Service) functions
* get_correlation_id (input: icn, output: correlation_id), self explanatory but some error handling about actual service failures or user not found
* get_verification_info (input: correlation_id, output: response hash parsed into attributes), some error handling about service failures or user not found, response hash parsed into a set of expected attributes that are easier to move around in Rails
 
`MHV::InheritedProofingVerifier` (Service Object) functions:
* new (input: user)
* perform (input: user, output: code), this would do the logic of determine if we need to extract correlation_id, whether or not the identity document exists on the verification response, saving to the redis, contextual error handling, etc

Reply:
I actually forgot that I hadn't done the Redis part when I said this was good to review. I'm planning on doing more or less what you described, which is creating a different service that does the work of determining whether redis is necessary etc